src/main/scala/beam/router/RouteHistory.scala

```diff
@@ -107,19 +107,26 @@ class RouteHistory @Inject()(
     }
   }
   override def notifyIterationEnds(event: IterationEndsEvent): Unit = {
-    val filePath = event.getServices.getControlerIO.getIterationFilename(
-      event.getServices.getIterationNumber,
-      RouteHistory.outputFileBaseName + ".csv.gz"
-    )
 
-    FileUtils.writeToFile(
-      filePath,
-      toCsv(routeHistory),
-    )
+    if (shouldWriteInIteration(event.getIteration, beamConfig.beam.physsim.writeRouteHistoryInterval)) {
+      val filePath = event.getServices.getControlerIO.getIterationFilename(
+        event.getServices.getIterationNumber,
+        RouteHistory.outputFileBaseName + ".csv.gz"
+      )
+
+      FileUtils.writeToFile(
+        filePath,
+        toCsv(routeHistory),
+      )
+    }
+
     previousRouteHistory = routeHistory
     routeHistory = new TrieMap()
   }
 
+  private def shouldWriteInIteration(iterationNumber: Int, interval: Int): Boolean = {
+    interval == 1 || (interval > 0 && iterationNumber % interval == 0)
```

Comment:
This will not write at all if it is a negative or 0. In an effort to handle misconfiguration maybe it should write every iteration in those cases?

Reply:
for negative we will do fail fast for this config param and for 0 we will write file in all iterations

Reply:
OK, I had been on mobile and see you added fail fast. And misread your 0 scenario - so approving